.github/workflows/release.yml

```
@@ -47,10 +47,18 @@ jobs:
         run: pnpm build
 
       - name: Create Release Pull Request or Publish to npm
+        id: release
         uses: changesets/action@v1
         with:
           title: Release tracking
           publish: npx changeset publish
```

```suggestion
          publish: npx changeset publish
          version: npx ../../scripts/changeset-version.mjs
```

And inside the mjs you run `changeset version` which gives you the version changes you need. So basically you use changesets CLI to do the version changes for you and append it to the compatibility.json file